docs/USAGE-GUIDE.md

```diff
@@ -485,7 +485,13 @@ Applications may want to know which certificate was used by a server for authent
 
 Additionally s2n-tls has functions for parsing certificate extensions on a certificate. Use `s2n_cert_get_x509_extension_value_length()` and `s2n_cert_get_x509_extension_value()` to obtain a specific DER encoded certificate extension from a certificate. `s2n_cert_get_utf8_string_from_extension_data_length()` and `s2n_cert_get_utf8_string_from_extension_data()` can be used to obtain a specific UTF8 string representation of a certificate extension instead. These functions will work for both RFC-defined certificate extensions and custom certificate extensions.
 
-### OCSP Stapling
+### Certificate Revocation
+
+Certificate revocation is a method in which CAs inform validators that an active certificate should not be trusted. This can occur when, for example, a private key has been leaked and the identity of the certificate's owner can no longer be trusted.
+
+s2n-tls supports two methods of certificate revocation validation: OCSP stapling, and CRLs. OCSP stapling differs from CRLs in that OCSP stapling is optional. A requester can ask a responder to send a stapled revocation status, but the responder may not actually send this response. However, CRLs exist for all valid certificates. If it can be assumed that OCSP stapling is supported by the peer of the application, it should be used as the method of certificate revocation validation. If this can not be assumed, and the CRLs for all received certificates can be obtained in advance, CRLs should be used instead.
```

Comment:
```suggestion
s2n-tls supports two methods of certificate revocation: OCSP stapling and CRLs. A fundamental difference between the two is that with OCSP stapling, the peer can choose not to send a certificate status response.  Applications using this method will have to decide whether or not to fail certificate validation in this case. In contrast, with CRLs the application can check the revocation status of any certificate encountered without relying on the peer. Users who want certificate revocation should look closely at their use-case and decide which method is more appropriate. We suggest using OCSP stapling if you're sure your peer supports OCSP stapling. Use CRLs if you're sure you can obtain CRLs for all certificates you expect to encounter.
```

I rewrote this to make the contrast between the two methods stronger.

Reply:
Should we mention that OCSP stapling requires a lot less data than CRLs?

Reply:
This is much clearer. Thank you!

I added a sentence about CRLs needing more memory/network bandwidth. I also split this up into two paragraphs (let me know if you prefer it as one, it just seemed really long).